app/legacy_lib/query_payments.rb

```diff
@@ -124,6 +132,9 @@ def self.full_search_expr(npo_id, query)
       expr = expr.order_by("NULLIF(payments.towards, '') #{query[:sort_towards]}")
     end
 
+    expr = expr.with(:nonprofits, get_nonprofit_query(npo_id), materialized: true)
```

Comment:
2 Thoughts on this: 
1. These lines of code have the same variable name. 
Can we rename the new variables to 'nonprofit_expr' and 'payment_expr'? 
2. If renaming is not viable, then will the computer recognize that each line of code has a value that contains a different query?

Reply:
Good question! We want it this way because of the API design of Qexpr. 

Qexpr objects are intentionally composable, a bit like ActiveRecord queries. On line 135, we're taking the expr from before this line, adding a nonprofits CTE `with` clause to that new object and then setting expr to that new object.

We're doing the same type of thing at the next line. It works a little bit like ActiveRecord queries which might be more familiar. Here's an example of how it might work with ActiveRecord:

```ruby
query = Nonprofit.where("name  ILIKE ?", "home%") # find all nonprofits whose name starts with home (regardless of case)
query = query.limit(3) # find up to 3 nonprofits whose name starts with home (regardless of case)
```

You can keep building your query object there, including by using conditional statements to add new conditions. But you keep the same working variable name so you always know which variable you should be adding a condition to.

Does that make sense?

Reply:
Yes! That makes sense. 
That's interesting that you can do that for Qexpr..
Thank you for explaining.  :)